src/viam/app/data/client.py

```diff
@@ -0,0 +1,100 @@
+from typing import List, Optional, Mapping, Any
+
+from grpclib.client import Channel
+
+from viam.proto.app.data import (
+    DataServiceStub,
+    Filter,
+)
+from viam.proto.app.datasync import (
+    UploadMetadata,
+    SensorData,
+    FileData
+)
+from viam.proto.app.datasync import DataSyncServiceStub
+from viam import logging
+
+LOGGER = logging.getLogger(__name__)
+
+
+class DataClient:
+    """gRPC client for uploading and retreiving data from app
+
+    Constructor must be passed a channel to app, along with the required metadata.
```

Comment:
Might be good to expand this comment a bit to indicate that using an `AppClient` is the preferred way to interact with a `DataClient`.